main.tf

```
@@ -1,25 +1,38 @@
 data "aws_caller_identity" "current" {
 }
 
+data "aws_region" "current" {
+}
+
 locals {
-  account_id = data.aws_caller_identity.current.account_id
+  account_id  = data.aws_caller_identity.current.account_id
+  bucket_name = var.bucket_name == null ? "${local.account_id}-${local.region}-s3logging-${var.bucket_suffix}" : var.bucket_name
```

I like the `coalesce()` pattern STY uses better for this
```suggestion

  bucket_name = coalesce(
    var.bucket_name,
    "${local.account_id}-${local.region}-s3logging-${var.bucket_suffix}"
  )

```